pypeit/core/flexure.py

```diff
@@ -303,147 +305,226 @@ def spec_flex_shift(obj_skyspec, arx_skyspec, arx_lines, mxshft=20):
         bpm (`numpy.ndarray`_):
             True = masked slit
-        method (:obj:`str`)
-          'boxcar' -- Recommneded
-          'slitpix' --
+        method (:obj:`str`, optional)
+          'boxcar' -- Recommended for object extractions
+          'slitcen' -- Recommended when no objects are being extracted
         sky_file (str):
             Sky file
+        specobjs (:class:`pypeit.specobjs.Specobjs`_, optional):
+            Spectral extractions
+        slitspecs (list, optional):
+            A list of linetools.xspectrum1d, one for each slit
+        waveimg (`numpy.ndarray`_, optional):
+            A 2D array providing the wavelength (in A) of each pixel of an image.
+            If none, the spectral flexure correction will not be applied to the
+            wavelength image. If provided, the correction will be applied and
+            returned.
         mxshft (int, optional):
             Passed to flex_shift()
+        slitShift (list, optional):
+            A list of spectral shifts (in pixels) that have been globally applied
+            to each slit and the waveimg. This list should have the same size as
+            the number of slits.
 
     Returns:
-        list:  list of dicts containing flexure results
-            Aligned with specobjs
-            Filled with a basically empty dict if the slit is skipped or there is no object
+        tuple: Two element tuple containing (1) the input waveimg (`numpy.ndarray`_),
+               corrected for spectral flexure and (2) a list of dicts containing flexure
+               results of each slit. The latter is filled with a basically empty dict if
+               the slit is skipped
     """
     sv_fdict = None
     msgs.work("Consider doing 2 passes in flexure as in LowRedux")
+    waveimg_flex_corr = None if waveimg is None else waveimg.copy()
+
+    # Determine the method
+    slit_cen = False
+    if specobjs is None: slit_cen = True
 
     # Load Archive. Save the line information to avoid the performance hit from calling it on the archive sky spectrum
     # multiple times
     sky_spectrum = load_sky_spectrum(sky_file)
     sky_lines = arc.detect_lines(sky_spectrum.flux.value)
 
-    nslits = len(bpm)
-    gdslits = np.where(np.invert(bpm))[0]
+    nslits = slits.nslits
+    gpm = np.logical_not(bpm)
+    gdslits = np.where(gpm)[0]
 
-    # Loop on objects
+    # Initialise the flexure list for each slit
     flex_list = []
-
     # Slit/objects to come back to
     return_later_sobjs = []
 
-    # Loop over slits, and then over objects here
+    # Loop over slits, and then over objects
     for islit in range(nslits):
-        i_slitord = slitord[islit]
-        msgs.info("Working on flexure in slit (if an object was detected): {:d}".format(islit))
+        msgs.info("Working on spectral flexure of slit: {:d}".format(islit))
 
-        indx = specobjs.slitorder_indices(i_slitord)
-        this_specobjs = specobjs[indx]
         # Reset
```

Comment:
Can we add a attribute to this dict indicating the method that was used, i.e. slits vs object?